net/packet/tsmp.go

```diff
@@ -129,12 +159,16 @@ func (pp *Parsed) AsTailscaleRejectedHeader() (h TailscaleRejectedHeader, ok boo
 	if len(p) < 7 || p[0] != byte(TSMPTypeRejectedConn) {
 		return
 	}
-	return TailscaleRejectedHeader{
+	h = TailscaleRejectedHeader{
 		Proto:  IPProto(p[1]),
 		Reason: TailscaleRejectReason(p[2]),
 		IPSrc:  pp.Src.IP,
 		IPDst:  pp.Dst.IP,
 		Src:    netaddr.IPPort{IP: pp.Dst.IP, Port: binary.BigEndian.Uint16(p[3:5])},
 		Dst:    netaddr.IPPort{IP: pp.Src.IP, Port: binary.BigEndian.Uint16(p[5:7])},
-	}, true
+	}
+	if len(p) > 7 {
+		h.MaybeBroken = (p[7] & 1) != 0
```

Comment:
Sounds like you're teeing this byte up to be a flag bitfield. If the wire format and parsed representation deviate further, it may be worth a longform comment to detail the wire format somewhere. Fine for now though.

Reply:
added some docs & consts & helper method to make the format & flags all clear